x-pack/test/functional/apps/canvas/reports.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const es = getService('es');
+  const esArchiver = getService('esArchiver');
+  const browser = getService('browser');
+  const log = getService('log');
+  const security = getService('security');
+  const PageObjects = getPageObjects(['reporting', 'common', 'canvas']);
+
+  describe('PDF Report Generation', () => {
+    before('initialize tests', async () => {
+      log.debug('ReportingPage:initTests');
+      await security.testUser.setRoles(['kibana_admin', 'reporting_user']);
+      await esArchiver.load('canvas/reports');
+      await browser.setWindowSize(1600, 850);
+    });
+    after('clean up archives', async () => {
+      await esArchiver.unload('canvas/reports');
+      await es.deleteByQuery({
+        index: '.reporting-*',
+        refresh: true,
+        body: { query: { match_all: {} } },
+      });
+    });
+
+    describe('Print PDF button', () => {
+      it('downloaded PDF base64 string is correct', async function () {
+        // Generating and then comparing reports can take longer than the default 60s timeout
+        this.timeout(180000);
+
+        await PageObjects.common.navigateToApp('canvas');
+        await PageObjects.canvas.loadSavedWorkpad('The Very Cool Workpad for PDF Tests');
+        await PageObjects.reporting.openPdfReportingPanel();
+        await PageObjects.reporting.clickGenerateReportButton();
+
+        const url = await PageObjects.reporting.getReportURL(60000);
+        const res = await PageObjects.reporting.getResponse(url);
+
+        expect(res.status).to.equal(200);
+        expect(res.get('content-length')).to.equal('1598');
+        expect(res.get('content-type')).to.equal('application/pdf');
+        expect(res.get('content-disposition')).to.equal(
+          'inline; filename="The Very Cool Workpad for PDF Tests.pdf"'
+        );
+
+        /* Check the value of the PDF data that was generated
+         * PDF files include dynamic meta info such as creation date.
+         * This checks only the first few thousand bytes of the Buffer
+         */
```

Comment:
This should probably use some dependency such as `compare-pdf` 

Comparing this buffer should be very simple - the PDF content is an 8 x 8 square of blue pixels. The differences will be in the PDF metadata.